workers/unity/Packages/com.improbable.gdk.mobile/Editor/iOSLaunchUtils.cs

```diff
@@ -0,0 +1,292 @@
+                {
+                    Debug.LogError("Was not able to find an XCode project. Did you build your iOS worker?");
+                    return;
+                }
+
+                if (string.IsNullOrEmpty(developmentTeamId))
+                {
+                    Debug.LogError("Development Team Id was not specified. Unable to build the XCode project.");
+                    return;
+                }
+
+                if (!TryBuildXCodeProject(developmentTeamId))
+                {
+                    Debug.LogError(
+                        $"Failed to build your XCode project. Make sure you have xcodebuild installed and check the logs.");
+                }
+            }
+            finally
+            {
+                EditorUtility.ClearProgressBar();
+            }
+        }
+        
+        public static void Launch(bool shouldConnectLocally, string deviceId, bool useSimulator)
+        {
+            try 
+            {
+                EditorUtility.DisplayProgressBar("Preparing your Mobile Client", "Preparing launch arguments", 0.0f);
+
+                if (!TryGetXCTestRunPath(useSimulator, out var xcTestRunPath))
+                {
+                    Debug.LogError(
+                        "Unable to find a xctestrun file for the correct architecture. Did you build your game using the correct Target SDK?");
+                    return;
+                }
+
+                if (!MobileLaunchUtils.TryPrepareArguments(shouldConnectLocally, out var arguments))
+                {
+                    Debug.LogError("Failed to generate arguments");
+                    return;
+                }
+                
+                if (!TryModifyEnvironmentVariables(xcTestRunPath, arguments))
+                {
+                    Debug.LogError($"Was unable to read and modify {xcTestRunPath}.");
+                    return;
+                }
+                
+                if (useSimulator)
+                {
+                    EditorUtility.DisplayProgressBar("Launching Mobile Client", "Start iOS Simulator", 0.5f);
+
+                    // Start simulator
+                    if (RedirectedProcess.Command("xcrun")
+                        .WithArgs("instruments", "-w", deviceId, "-t", "Blank")
+                        .Run() != 0)
+                    {
+                        Debug.LogError("Was unable to start iOS Simulator.");
+                        return;
+                    }
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Installing your app", 0.7f);
+
+                if (!TryLaunchApplication(deviceId, xcTestRunPath))
+                {
+                    Debug.LogError("Failed to start app on iOS device.");
+                }
+
+                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Done", 1.0f);
+            }
+            finally
+            {
+                var traceDirectories = Directory
```

Comment:
Why do we do this?

Reply:
because we need to start the simulator with a trace template (otherwise the process will return a non-0 error code) which will create a `.trace` file whenever we start the simulator. 
I still need to look more into on how the trace could be used, but as we simply define a blank template I doubt that it contains a lot of information.